### PR TITLE
Ensure the value of default option must be a string when calling I18n.t

### PR DESCRIPTION
**Pull request by mlzhuyi**
https://github.com/ruby-grape/grape/issues/1735

### CODE REVIEW

File: lib/grape/exceptions/base.rb [hunk 1]
```diff
@@ -51,12 +51,12 @@ def resolution(key, attributes)
...
       def translate_attributes(keys, **options)
         keys.map do |key|
-          translate("#{BASE_ATTRIBUTES_KEY}.#{key}", default: key, **options)
+          translate("#{BASE_ATTRIBUTES_KEY}.#{key}", default: key.to_s, **options)
         end.join(', ')
       end
 
       def translate_attribute(key, **options)
-        translate("#{BASE_ATTRIBUTES_KEY}.#{key}", default: key, **options)
+        translate("#{BASE_ATTRIBUTES_KEY}.#{key}", default: key.to_s, **options)
```
Comment (dblock) [inline lib/grape/exceptions/base.rb]: Maybe I am nitpicking, but feels like we should do this inside `translate`, no?

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,7 @@
 #### Fixes
 
 * Your contribution here.
+* [#1737](https://github.com/ruby-grape/grape/pull/1737): Ensure the value of default option must be a string when calling i18n.t - [@mlzhuyi](https://github.com/mlzhuyi).
```
Comment (dblock) [inline CHANGELOG.md]: This fixes a bug. What's the error that you get without this fix? That's what should be in the changelog, please.

File: lib/grape/exceptions/base.rb (hunk 1 above)
Comment (mlzhuyi) [reply lib/grape/exceptions/base.rb]: It is generally better. But I'm not sure if there is any need to pass an alternative symbol key as value of :default in other cases, though, there isn't any example now.

Comment (mlzhuyi) [reply lib/grape/exceptions/base.rb]: It is generally better. But I'm not sure if there is any need to pass an alternative symbol key as value of :default in other cases, though, there isn't any example now.

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,7 @@
 #### Fixes
 
 * Your contribution here.
+* [#1737](https://github.com/ruby-grape/grape/pull/1737): Fix translating error whe pass symbols as params in custom validations - [@mlzhuyi](https://github.com/mlzhuyi).
```
Comment (dblock) [inline CHANGELOG.md]: Typo: `when passing`

**Comment by mlzhuyi** [comment]
It has been changed

File: lib/grape/exceptions/base.rb [hunk 4]
```diff
@@ -71,6 +71,7 @@ def translate_message(key, **options)
       end
 
       def translate(key, **options)
+        options[:default] &&= options[:default].to_s
```
Comment (dblock) [inline lib/grape/exceptions/base.rb]: This modifies the caller's `options`, which can easily be a frozen hash or whatever.

Comment (mlzhuyi) [reply lib/grape/exceptions/base.rb]: `*` and `**` in arguments act a different way with common argument.

```ruby
def common_argument_test(args)
  p args.object_id
end

def keyword_argument_test(**args)
  p args.object_id
end

args = { a: 1 }
p args.object_id # => 70338004137000
common_argument_test args # => 70338004137000
keyword_argument_test args # => 70338004136780
```
In there code, `args` in `keyword_argument_test ` is a new hash, create by the method itself.I think we can treat it as a most basic hash.

Comment (dblock) [reply lib/grape/exceptions/base.rb]: I guess we always declared it the old fashioned way as `options = {}`. I worry that this is different version from version, especially reading [this SO](https://stackoverflow.com/questions/18289152/what-does-a-double-splat-operator-do). Can we just do this the old fashioned way and avoid surprises?

Comment (mlzhuyi) [reply lib/grape/exceptions/base.rb]: Do you mean refactoring out the code, for there here are many `**` now?

Comment (dblock) [reply lib/grape/exceptions/base.rb]: Sorry if I wasn't clear. The double splat is fine. I mean this:

```ruby
        options = options.dup
        options[:default] &&= options[:default].to_s
```

We dup arguments everywhere as far as I can see.

Comment (mlzhuyi) [reply lib/grape/exceptions/base.rb]: It's a really good habit in case of unknown changes. <br>I have added it :)

**Comment by dblock** [comment]
Thanks, merged.

**Pull request merged**